plugins/hls-pragmas-plugin/src/Ide/Plugin/Pragmas.hs

```diff
@@ -214,59 +186,70 @@ completion _ide _ complParams = do
-                      }
         _ -> return $ J.List []
 
 -----------------------------------------------------------------------
-validPragmas :: [(T.Text, T.Text, T.Text)]
+
+-- | Pragma where exist
+data AppearWhere =
+  NewLine
+  -- ^Must be on a new line
+  | CanInline
+  -- ^Can appear in the line
+  deriving (Show, Eq)
+
+validPragmas :: [(T.Text, T.Text, T.Text, AppearWhere)]
 validPragmas =
-  [ ("LANGUAGE ${1:extension}"         , "LANGUAGE",           "{-# LANGUAGE #-}")
-  , ("OPTIONS_GHC -${1:option}"        , "OPTIONS_GHC",        "{-# OPTIONS_GHC #-}")
-  , ("INLINE ${1:function}"            , "INLINE",             "{-# INLINE #-}")
-  , ("NOINLINE ${1:function}"          , "NOINLINE",           "{-# NOINLINE #-}")
-  , ("INLINABLE ${1:function}"         , "INLINABLE",          "{-# INLINABLE #-}")
-  , ("WARNING ${1:message}"            , "WARNING",            "{-# WARNING #-}")
-  , ("DEPRECATED ${1:message}"         , "DEPRECATED",         "{-# DEPRECATED  #-}")
-  , ("ANN ${1:annotation}"             , "ANN",                "{-# ANN #-}")
-  , ("RULES"                           , "RULES",              "{-# RULES #-}")
-  , ("SPECIALIZE ${1:function}"        , "SPECIALIZE",         "{-# SPECIALIZE #-}")
-  , ("SPECIALIZE INLINE ${1:function}" , "SPECIALIZE INLINE",  "{-# SPECIALIZE INLINE #-}")
+  [ ("LANGUAGE ${1:extension}"        , "LANGUAGE"         ,  "{-# LANGUAGE #-}"        ,   NewLine)
+  , ("OPTIONS_GHC -${1:option}"       , "OPTIONS_GHC"      ,  "{-# OPTIONS_GHC #-}"     ,   NewLine)
+  , ("INLINE ${1:function}"           , "INLINE"           ,  "{-# INLINE #-}"          ,   NewLine)
+  , ("NOINLINE ${1:function}"         , "NOINLINE"         ,  "{-# NOINLINE #-}"        ,   NewLine)
+  , ("INLINABLE ${1:function}"        , "INLINABLE"        ,  "{-# INLINABLE #-}"       ,   NewLine)
+  , ("WARNING ${1:message}"           , "WARNING"          ,  "{-# WARNING #-}"         , CanInline)
+  , ("DEPRECATED ${1:message}"        , "DEPRECATED"       ,  "{-# DEPRECATED  #-}"     , CanInline)
+  , ("ANN ${1:annotation}"            , "ANN"              ,  "{-# ANN #-}"             ,   NewLine)
+  , ("RULES"                          , "RULES"            ,  "{-# RULES #-}"           ,   NewLine)
+  , ("SPECIALIZE ${1:function}"       , "SPECIALIZE"       ,  "{-# SPECIALIZE #-}"      ,   NewLine)
+  , ("SPECIALIZE INLINE ${1:function}", "SPECIALIZE INLINE", "{-# SPECIALIZE INLINE #-}",   NewLine)
+  , ("SPECIALISE ${1:function}"       , "SPECIALISE"       , "{-# SPECIALISE #-}"       ,   NewLine)
+  , ("SPECIALISE INLINE ${1:function}", "SPECIALISE INLINE", "{-# SPECIALISE INLINE #-}",   NewLine)
+  , ("MINIMAL ${1:functions}"         , "MINIMAL"          ,  "{-# MINIMAL #-}"         , CanInline)
+  , ("LINE ${1:#line} ${2:file}"      , "LINE"             , "{-# LINE #-}"             , CanInline)
+  , ("COLUMN ${1:#column}"            , "COLUMN"           , "{-# COLUMN #-}"           , CanInline)
```

Comment:
I'm not sure we want to offer completions for these pragmas, as we don't expect users to ever introduce them, right?